src/vm/comutilnative.cpp

```diff
@@ -1288,6 +1289,49 @@ FCIMPL3(Object*, GCInterface::AllocateNewArray, void* arrayTypeHandle, INT32 len
 }
 FCIMPLEND
 
+
+FCIMPL1(INT64, GCInterface::GetTotalAllocatedBytes, CLR_BOOL precise)
+{
+    FCALL_CONTRACT;
+
+    if (!precise)
+    {
+        // we do not want to suspend the EE for imprecise flavor, 
+        // but dead_threads_non_alloc_bytes is 64bit and may be concurrently updated.
+        // ensure atomic read on 32 bit
+#ifdef _TARGET_64BIT_
+        uint64_t unused_bytes = Thread::dead_threads_non_alloc_bytes;
+#else
+        uint64_t unused_bytes = FastInterlockExchangeAddLong((LONG64*)&Thread::dead_threads_non_alloc_bytes, 0);
```

Comment:
Not completely sure we need to do this. There is a possibility that on 32bit we read a torn 64bit value.
The value is monotonically increasing. I suppose when crossing 32bit value we can report a total that is temporarily jumps/drops by 2^32 . 
That should be rare, but could in theory violate the invariant "dirty value is always higher than precise one taken earlier"